### PR TITLE
mdbook initial setup

### PR DESCRIPTION
**Pull request by kitzbergerg**
# Introduction

Add mdbook: `cargo install mdbook`.  
You can now run `mdbook serve --open` to generate documentation from markdown files in the doc folder.

## Basics

- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

File: doc/SUMMARY.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Summary
+
+- [architecture](./architecture/README.md)
+  - [context](./architecture/context.md)
+  - [database_schemata](./architecture/database_schemata.md)
+  - [glossary](./architecture/glossary.md)
+  - [sequence](./architecture/sequence.md)
+  - [solution](./architecture/solution.md)
+- [authors](./authors.md)
```
Comment (markus2330) [inline doc/SUMMARY.md:9]: Can be much later/last.

File: doc/SUMMARY.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+- [ci](./ci.md)
+- [contrib](./contrib/README.md)
+  - [backend](./contrib/backend.md)
+  - [frontend](./contrib/frontend.md)
+- [decisions](./decisions/README.md)
```
Comment (markus2330) [inline doc/SUMMARY.md:14]: Do we really have to list each one of them? Can we somehow include the whole folder?

File: doc/SUMMARY.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  - [frontend_build_tool](./decisions/frontend_build_tool.md)
+  - [frontend_programming_language](./decisions/frontend_programming_language.md)
+  - [frontend_state_management](./decisions/frontend_state_management.md)
+  - [frontend_ui_framework](./decisions/frontend_ui_framework.md)
+  - [reversible_deletion](./decisions/reversible_deletion.md)
+- [development_setup](./development_setup.md)
+- [documentation](./documentation.md)
+- [install](./install.md)
+- [meetings](./meetings/README.md)
+  - [2022_12_16](./meetings/2022_12_16.md)
```
Comment (markus2330) [inline doc/SUMMARY.md:34]: Same question: Can we somehow include the whole folder?

File: doc/SUMMARY.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
+# Summary
+
```
Comment (markus2330) [inline doc/SUMMARY.md:2]: Probably main README.md should be first

File: book.toml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+[book]
+authors = ["kitzbergerg"]
```
Comment (markus2330) [inline book.toml:2]: Can you link to doc/authors.md here? Otherwise write something generic like "PermaplanT Contributors"

**Review by markus2330 — APPROVED** [review]
Thank you for the PR! Please write in the documentation how the mdbook can be built.

**Comment by markus2330** [comment]
I think we also want (from https://github.com/rust-lang/mdBook/wiki/Third-party-plugins)

- maybe toc or [pagetoc](https://github.com/JorelAli/mdBook-pagetoc) for meetings+use cases to not make the left side bar too long
- mermaid
- [linkcheck](https://github.com/Michael-F-Bryan/mdbook-linkcheck)

**Comment by kitzbergerg** [comment]
Resolved all the threads above as they will be resolved by automatic generating.
 
> Maybe someone already wrote the possibility to include a whole folder? If not, can you open an issue at their tracker, maybe they want such a feature anyway.
 I found [this](https://github.com/cococolanosugar/mdbook-auto-gen-summary). A Rust pre-processor doing exactly that. I can test it and if we need more functionality I would suggest forking it or creating our own.
Writing our own might take a few hours so probably a individual task for 1 week.

**Comment by kitzbergerg** [comment]
@markus2330 It is now possible to use the preprocessor. To do that run:
```
cargo install --version 0.1.0 --git https://github.com/kitzbergerg/mdbook-generate-summary mdbook-generate-summary
mdbook serve --open
```

The cargo command installs the preprocessor as a binary (I put it in a separate repo as it doesn't really fit here). 
mdbook should now automatically generate all the chapters so from now on we just need to put them in the correct order.

File: doc/architecture/README.md [hunk 6]
```diff
@@ -0,0 +1 @@
+# architecture
```
Comment (markus2330) [inline doc/architecture/README.md:1]: Should all be capitalized. Please also write a bit about it, e.g. that we use arc42, have a frontend+backend etc.

`1.5. ## Solution` there is some problem, please check the titles.

```suggestion
# Architecture
```

File: doc/meetings/README.md [hunk 7]
```diff
@@ -0,0 +1 @@
+# meetings
```
Comment (markus2330) [inline doc/meetings/README.md:1]:
```suggestion
# Meetings
```

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great, works for me! Can't wait to get this, very nice way to read documentation.

Please add **within the repository** what exactly needs to be done and please don't write it in the PR description as it may lead to confusion in which directory something actually should be executed. Also `cargo install mdbook` is missing.

**Comment by markus2330** [comment]
removed this from project and added issue instead

**Comment by kitzbergerg** [comment]
@markus2330 I released a new version of the preprocessor. To build and view the doc run:
```
cargo install mdbook
cargo install --version 0.1.1 --git https://github.com/kitzbergerg/mdbook-generate-summary mdbook-generate-summary
mdbook serve --open
```

I still have to document the other repo, but apart from that this PR is pretty much done. I would leave the rest to @lukashartl (I'll create a new issue describing what to do).

The question now is pretty much how we want to continue with this. We could either merge it like this and gradually add explanations and references (e.g. were to find backend doc, how to build with mdbook) or make it part of this PR as well. 

One 'breaking change' is that as soon as this PR is merged every directory in doc is required to include a `README.md`. There is the option to disable that in the preprocessor, but I think we should enforce a short explanation as to which directory does what.

**Review by markus2330 — CHANGES_REQUESTED** [review]
There is still no documentation **within the repository** how to run the mdbook. Please add it to the top-level README.md

**Comment by markus2330** [comment]
Yes, I agree, all folders should have a README.md

File: README.md [hunk 8]
```diff
@@ -5,6 +5,19 @@ PermaplanT is an app for
...
 - Larger mobile devices like tablets: Progressive Web App (PWA) Android 9+
 
+## Documentation
+
+Relevant documentation for this initiative can be found in the folder `doc`.  
+You can also build and view the documentation in your browser by running the following commands in the projects root folder:
+
+```sh
+cargo install mdbook
+cargo install --version 0.1.1 --git https://github.com/kitzbergerg/mdbook-generate-summary mdbook-generate-summary
```
Comment (markus2330) [inline README.md:15]: For me this compiled and installed mdbook v0.4.28 again?

**Pull request merged**